README.md

```diff
@@ -65,6 +66,19 @@ Compile the contracts and run tests
 yarn testc
 ```
 
+## Code Formatting
+
+This repository uses [Prettier](https://prettier.io/) for code formatting with configuration in `.prettierrc`. Formatting is automatically enforced using [Husky](https://typicode.github.io/husky/) and [lint-staged](https://github.com/lint-staged/lint-staged) to ensure consistent code style across all contributions.
+
+### Available Commands
+
+- Format all files: `yarn format`
+- Check formatting without modifying files: `yarn format:check`
+
+### Automatic Formatting
+
+Files are automatically formatted when you commit changes. The pre-commit hook will run Prettier on staged files before they are committed, ensuring that all code in the repository follows the same formatting standards.
+
```

Comment:
yarn formatting works, however was still able to commit unformatted files 

git add . && git commit -m "some message"

Will take a deeper look later

Reply:
are you sure?

**1. Before**
![image](https://github.com/user-attachments/assets/04ef35d7-aa6d-4434-9be6-eb03eb3e6203)

---------

**2.  Run `git add . && git commit -m "some message"`**

![image](https://github.com/user-attachments/assets/4d05b66f-8f49-4666-91f4-0458ede201d6)

---------

**3. After**

![image](https://github.com/user-attachments/assets/87443164-76b7-4f24-accd-0285be934be1)

---------

Reply:
I've modified a file:
<img width="1271" alt="image" src="https://github.com/user-attachments/assets/ac622e94-e9be-48ad-b6c1-dee81d15d148" />

stage & commit:
<img width="1140" alt="image" src="https://github.com/user-attachments/assets/4dec8770-a6a7-437c-84bf-c11a183ed248" />

file is the same

Reply:
Running yarn format after comitting:
<img width="1186" alt="image" src="https://github.com/user-attachments/assets/0f4f854e-9849-4619-8d24-c4eacaa7a670" />

Reply:
so npx lint-staged fails if it would result in an empty commit (when there are no changes, apart from formating) 
<img width="511" alt="image" src="https://github.com/user-attachments/assets/b738f834-d814-4dcd-b965-db6656853f88" />

When there are changes npx lint-staged does format it and work, but it doesn't trigger for me automatically on commit, I have to explicitly run it - not sure why, will investigate further

Reply:
Ok so husky hooks weren't triggering - I've pushed a fix